### ISSUE TITLE
Mismatch between origin and baseUrl (dev)

### ISSUE THREAD

**Issue by omarcostahamido** [hosso/notion-custom-domain#22]
@ShiinaRinne @hosso 
Since we can't reopen #17 here is a new issue. Please refer to the final discussion over there for more context.

>It seems that Notion has written some things in a new mainApp-*.js.

>We tried to apply the rules of app-*.js, but did not get the expected results.

**Comment by omarcostahamido** [comment]
I am a bit sad that we can't fix this one... 'twas good while it lasted though!

**Comment by hosso** [comment]
Unfortunately, it may take a lot of time to analyze it.

**Comment by omarcostahamido** [comment]
indeed, i lost several days on it and could not find the solution 😞

**Comment by mrothko2** [comment]
Any chance we can get this fixed before Christmas? :)

**Comment by velsa** [comment]
Hi everyone!

I just published an npm package [notehost](https://www.npmjs.com/package/notehost), which allows you to 🌟 host your Notion page for free on any domain 🌟

This is a CLI and an SDK with which you can create a worker on Cloudflare and configure it on your domain. The code is completely open source. It is loosely based on Fruition, but I rewrote it in Typescript and added a CLI to make it even easier for people to configure and publish their Notion pages.

All critique and comments are welcome 🙏

**Comment by ShiinaRinne** [comment]
> Hi everyone!
> 
> I just published an npm package [notehost](https://www.npmjs.com/package/notehost), which allows you to 🌟 host your Notion page for free on any domain 🌟
> 
> This is a CLI and an SDK with which you can create a worker on Cloudflare and configure it on your domain. The code is completely open source. It is loosely based on Fruition, but I rewrote it in Typescript and added a CLI to make it even easier for people to configure and publish their Notion pages.
> 
> All critique and comments are welcome 🙏

Thanks for your great work! 

However, I have some questions:
- It seems that both Fruition and Notehost do not support `Comment` or `Duplicate`. Is there a way to fix this?
- Since access is too slow when using workers, is there any way to deploy it on Vercel?

I apologize for my lack of frontend experience and am very unfamiliar with these technical principles.

**Comment by velsa** [comment]
> > Hi everyone!
> > I just published an npm package [notehost](https://www.npmjs.com/package/notehost), which allows you to 🌟 host your Notion page for free on any domain 🌟
> > This is a CLI and an SDK with which you can create a worker on Cloudflare and configure it on your domain. The code is completely open source. It is loosely based on Fruition, but I rewrote it in Typescript and added a CLI to make it even easier for people to configure and publish their Notion pages.
> > All critique and comments are welcome 🙏
> 
> Thanks for your great work!
> 
> However, I have some questions:
> 
> * It seems that both Fruition and Notehost do not support `Comment` or `Duplicate`. Is there a way to fix this?

Not sure what you mean by "comment" and "duplicate".
What notehost essentially does is it simply proxies your public notion pages through your domain. So whatever is working/not working on the original public page should be working/not working on notehost. There might be a few exceptions (like search or login), but in general most page elements are simply proxied as they are.

> * Since access is too slow when using Cloudflare workers in China, is there any way to deploy it on Vercel?
> 

It should be possible to deploy notehost on Vercel or any other cloud provider, but you will have to learn more about nodejs, domains and dev ops. I will try to find some time to create a tutorial for Vercel, but it's not something I can promise to do in the near future.

> I apologize for my lack of frontend experience and am very unfamiliar with these technical principles. I hope to get your help~

**Comment by ShiinaRinne** [comment]
> > * It seems that both Fruition and Notehost do not support `Comment` or `Duplicate`. Is there a way to fix this?似乎Fruition和Notehost都不支持 `Comment` 或 `Duplicate` 。有办法解决这个问题吗？
> 
> Not sure what you mean by "comment" and "duplicate".不知道你说的“评论”和“复制”是什么意思。 What notehost essentially does is it simply proxies your public notion pages through your domain. So whatever is working/not working on the original public page should be working/not working on notehost. There might be a few exceptions (like search or login), but in general most page elements are simply proxied as they are. notehost本质上做的是它只是通过你的域代理你的公共概念页面。因此，无论在原始公共页面上工作/不工作，都应该在notehost上工作/不工作。可能会有一些例外（比如search或login），但一般来说，大多数页面元素都是简单地代理它们。

![image](https://github.com/hosso/notion-custom-domain/assets/45081750/0e9f9f1f-7dc5-4836-ab22-685f4c287d30)

It is similar to login, we need to log in before we can use it. 
@hosso 's [notion-custom-domain](https://github.com/hosso/notion-custom-domain) could support `login, comment` and other functions (I remember it would be redirected to the xx.notion.site page), 

But when using `Fruition`, it will cause an error request.
https://github.com/ShiinaRinne/fruitionsite
Therefore, I once modified `Fruition` to directly redirect to `xx.notion.site` through a button, and then let the viewer use the `comment` feature

**Comment by velsa** [comment]
`xx.notion.site` redirection is supported in the latest version of notehost. 

Just set `notionDomain` in your config file.

**Comment by hosso** [comment]
I've performed maintenance to make it work with the latest Notion at 8fec7c7daa811b956cb49e13c6346d45323efe96. [My demo site](https://notion-custom-domain.hosso.co) has also been restored.

**Comment by omarcostahamido** [comment]
thanks @hosso 😄
